splito/_split.py

```diff
@@ -0,0 +1,101 @@
+from enum import Enum, unique
+from typing import Optional, Sequence, Union
+
+import datamol as dm
+import numpy as np
+from sklearn.model_selection import ShuffleSplit
+
+from ._distribution_split import StratifiedDistributionSplit
+from ._kmeans_split import KMeansSplit
+from ._max_dissimilarity_split import MaxDissimilaritySplit
+from ._min_max_split import MolecularMinMaxSplit
+from ._molecular_weight import MolecularWeightSplit
+from ._perimeter_split import PerimeterSplit
+from ._scaffold_split import ScaffoldSplit
+
+
+@unique
+class SimpleSplittingMethod(Enum):
+    RANDOM = ShuffleSplit
+    KMEANS = KMeansSplit
+    PERIMETER = PerimeterSplit
+    MAX_DISSIMILARITY = MaxDissimilaritySplit
+    SCAFFOLD = ScaffoldSplit
+    STRATIFIED_DISTRIBUTION = StratifiedDistributionSplit
+    MOLECULAR_WEIGHT = MolecularWeightSplit
+    MIN_MAX_DIVERSITY_SPLIT = MolecularMinMaxSplit
+
+
+def train_test_split(
+    X: np.ndarray,
+    y: np.ndarray,
+    molecules: Optional[Sequence[Union[str, dm.Mol]]] = None,
+    method: Union[str, SimpleSplittingMethod] = "random",
```

Comment:
No support for custom callables for the splitting method?
Not sure if that defeats the purpose of `Splito`

Reply:
I think that defeats the purpose of Splito. If users request it, I would be happy to consider ways of adding this, but for now I think it's best to minimize complexity.